### PR TITLE
fix: safe access navigation entry type

### PR DESCRIPTION
**Pull request by AndersDJohnson**
Closes https://github.com/GoogleChrome/web-vitals/issues/291.

Work in progress...

We could also consider sending `undefined` instead of `'unknown'`.

Not sure how this new value affects the rest of the system, especially the types.

### CODE REVIEW

File: src/lib/initMetric.ts [hunk 1]
```diff
@@ -32,9 +32,11 @@ export const initMetric = (name: Metric['name'], value?: number): Metric => {
       navigationType = 'prerender';
     } else if (document.wasDiscarded) {
       navigationType = 'restore';
-    } else {
+    } else if (navEntry.type) {
       navigationType =
           navEntry.type.replace(/_/g, '-') as Metric['navigationType'];
+    } else {
+       navigationType = 'unknown';
```
Comment (AndersDJohnson) [inline src/lib/initMetric.ts:39]: TODO: Type here?

**Comment by AndersDJohnson** [comment]
@tunetheweb Done, thanks!

**Review by tunetheweb — APPROVED** [review]
LGTM!

**Comment by cklim4405** [comment]
> Closes https://github.com/GoogleChrome/web-vitals/issues/291.
> 
> Work in progress...
> 
> We could also consider sending `undefined` instead of `'unknown'`.
> 
> Not sure how this new value affects the rest of the system, especially the types.
>

**Pull request merged**